iohk-monitoring/src/Cardano/BM/Data/Severity.lhs

```diff
@@ -54,6 +54,31 @@ data Severity = Debug
               | Emergency
                 deriving (Show, Eq, Ord, Bounded, Enum, Generic, ToJSON, Read)
 
+\end{code}
+
+|Severity| is a \href{https://www.wikiwand.com/en/Semilattice}{lower
+semilattice}, and thus a monoid:
+\begin{code}
+instance Semigroup Severity where
+  Debug     <> _         = Debug
+  _         <> Debug     = Debug
+  Info      <> _         = Info
+  _         <> Info      = Info
+  Notice    <> _         = Notice
+  _         <> Notice    = Notice
+  Warning   <> _         = Warning
+  _         <> Warning   = Warning
+  Error     <> _         = Error
+  _         <> Error     = Error
+  Critical  <> _         = Critical
+  _         <> Critical  = Critical
+  Alert     <> _         = Alert
+  _         <> Alert     = Alert
+  Emergency <> Emergency = Emergency
```

Comment:
I agree with adding the monoid instance.

I'm not sure I agree that we could remove ord. We could only remove Ord if we provide and use threshold-testing functions everywhere.

Reply:
> I'm not sure I agree that we could remove ord. We could only remove Ord if we provide and use threshold-testing functions everywhere.

If it is too difficult to remove it, because it appears in other types that need to have `Ord` instances or otherwise,  we could use something like [Ordered](https://hackage.haskell.org/package/lattices-2.0.2/docs/Algebra-Lattice-Ordered.html#t:Ordered) newtype wrapper.  At least one needs to be explicit to use the `Ord` instance.

Reply:
I'm somewhat confused about what makes `Ord` confusing -- and what would make the `Ordered` less confusing?

`Ord` provides an ordering, that is used by `>`/`<`.  

`Ordering` would also provide an ordering, and would also come together with functions with similar meaning.

Where is the explanatory benefit?

Reply:
Threshold-testing can be done in this way:
```
atLeastWarning a = a <> Warning == Warning
```
no need for `Ord` to do that.  By the way, in lattice theory this the way to show that every semi lattice is a partial order (with finite infima) and vice versa if I recall correctly.